torch/distributed/optim/functional_adamax.py

```diff
@@ -100,4 +102,5 @@ def step(self, gradients: List[Optional[Tensor]]):
                      beta1=self.defaults['beta1'],
                      beta2=self.defaults['beta2'],
                      lr=self.defaults['lr'],
-                     weight_decay=self.defaults['weight_decay'])
+                     weight_decay=self.defaults['weight_decay'],
+                     foreach=self.foreach)
```

Comment:
cc @mrshenli are these objects ever serialized? Should we take extra care to make sure an old saved version still works?